base_local_planner/src/map_grid_cost_function.cpp

```diff
@@ -111,6 +112,11 @@ double MapGridCostFunction::scoreTrajectory(Trajectory &traj) {
       }
     }
 
+    if (firstCost < 0)
+    {
+      firstCost = grid_dist;
```

Comment:
Are we only using the first cost because we are using the DWA planner?  Or should we take the max cost?